scripts/gb-model/create_region_shapes.py

```diff
@@ -0,0 +1,421 @@
+
+logger = logging.getLogger(__name__)
+
+
+def load_country_shapes(filepath):
+    """
+    Load country shapes from GeoJSON file.
+
+    Args:
+        filepath (str): Path to the country shapes GeoJSON file
+
+    Returns:
+        geopandas.GeoDataFrame: Country shapes data
+    """
+    try:
+        logger.info(f"Loading country shapes from: {filepath}")
+        country_gdf = gpd.read_file(filepath)
+        logger.info(f"Loaded {len(country_gdf)} country shapes")
+        return country_gdf
+    except Exception as e:
+        logger.error(f"Error loading country shapes: {e}")
+        raise
+
+
+def load_boundary_lines(filepath):
+    """
+    Load boundary lines from shapefile or GeoJSON.
+
+    Args:
+        filepath (str): Path to the boundary lines file
+
+    Returns:
+        geopandas.GeoDataFrame: Boundary lines data
+    """
+    try:
+        logger.info(f"Loading boundary lines from: {filepath}")
+        boundary_gdf = gpd.read_file(filepath)
+        logger.info(f"Loaded {len(boundary_gdf)} boundary features")
+        return boundary_gdf
+    except Exception as e:
+        logger.error(f"Error loading boundary lines: {e}")
+        raise
+
+
+def ensure_same_crs(gdf1, gdf2):
+    """
+    Ensure both GeoDataFrames have the same CRS.
+
+    Args:
+        gdf1, gdf2 (geopandas.GeoDataFrame): GeoDataFrames to align
+
+    Returns:
+        tuple: Both GeoDataFrames with the same CRS
+    """
+    if gdf1.crs != gdf2.crs:
+        logger.info(f"Converting CRS from {gdf2.crs} to {gdf1.crs}")
+        gdf2 = gdf2.to_crs(gdf1.crs)
+    return gdf1, gdf2
+
+
+def create_regions_from_boundaries(country_shapes, boundary_lines):
+    """
+    Create regions by dividing country shapes using boundary lines.
+
+    Args:
+        country_shapes (geopandas.GeoDataFrame): Country polygons
+        boundary_lines (geopandas.GeoDataFrame): Boundary lines for division
+
+    Returns:
+        geopandas.GeoDataFrame: Regional divisions
+    """
+    logger.info("Creating regions from boundaries...")
+
+    # Convert to a projected CRS for accurate area calculations
+    # Use British National Grid (EPSG:27700) which is appropriate for UK
+    target_crs = "EPSG:27700"
+    logger.info(f"Converting to projected CRS {target_crs} for accurate measurements")
+
+    if country_shapes.crs != target_crs:
+        country_shapes = country_shapes.to_crs(target_crs)
+        logger.info(
+            f"Country shapes converted. New total area: {country_shapes.geometry.area.sum() / 1000000:.0f} km²"
+        )
+    if boundary_lines.crs != target_crs:
+        boundary_lines = boundary_lines.to_crs(target_crs)
+        logger.info(
+            f"Boundary lines converted. Length range: {boundary_lines.geometry.length.min():.0f} - {boundary_lines.geometry.length.max():.0f} meters"
+        )
+
+    if len(boundary_lines) == 0:
+        logger.warning("No valid boundary lines found!")
+        return country_shapes.copy()
```

Comment:
replace with an error message that the function is expecting a non-empty bounary_lines dataframe